src/svc_auth_gss.c

```diff
@@ -146,12 +146,13 @@ svcauth_gss_acquire_cred(void)
 				svcauth_gss_creds_expires = 0;
 			else
 				svcauth_gss_creds_expires = now + timerec;
-			if (old_creds) {
-				svcauth_prev_gss_creds = old_creds;
-			}
-			if (ancient_creds) {
-				(void) gss_release_cred(&min_stat, &ancient_creds);
-			}
+		}
+		/* release the old creds anyway */
```

Comment:
It is not clear from the available documentation that on failure of gss_acquire_cred(), it will hold the previous value. So considering it may return NULL or any other garbage, its better to handle the cred free job outside of the success block.
As copying of old_creds & ancient_creds is done irrespective of success to the gss_acquire_cred() routine, its more logical to handle the free job outside of the success block.
I do not think there will be any double free. Only the call to gss_release_cred will fail with return code GSS_S_NO_CRED.
But if the creds are freed only on success of gss_acquire_cred(), then we end up leaking old_creds if svcauth_gss_creds do not point to the previous cred handle (But it is not cleraly documented either way)

Reply:
Not changing this part in the updated patch